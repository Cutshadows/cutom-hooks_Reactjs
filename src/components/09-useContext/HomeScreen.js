import React, { useState, useContext } from 'react';
import UserContext from './userContext';


const HomeScreen = () => {
    const {user} = useContext(UserContext);

    return ( 
        <>
            <h1>HomeScreen</h1>

            <pre >
                {JSON.stringify(user, null, 3)}
            </pre>
        </>
     );
}
 
export default HomeScreen;