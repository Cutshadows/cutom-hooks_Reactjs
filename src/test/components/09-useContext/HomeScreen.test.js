import { shallowm, mount } from 'enzyme';
import React from 'react';
import HomeScreen from '../../../components/09-useContext/HomeScreen';
import UserContext from '../../../components/09-useContext/userContext';


describe('Pruebas sobre <HomeScreen></HomeScreen>', () => {
    
    const user={
        name:'Fernando',
        email:'Fernando@gmail.com'
    }
    const wrapper=mount(
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen />
        </UserContext.Provider>    
    )
    test('debe de cargar el componente ', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
})
