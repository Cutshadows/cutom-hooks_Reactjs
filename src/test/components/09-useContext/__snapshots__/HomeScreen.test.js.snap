// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <HomeScreen></HomeScreen> debe de cargar el componente  1`] = `
Array [
  <h1>
    HomeScreen
  </h1>,
  <pre>
    {
   "name": "Fernando",
   "email": "Fernando@gmail.com"
}
  </pre>,
]
`;
